### PR TITLE
marwaita: init at 2020-07-01

### PR DESCRIPTION
**Pull request by romildo**
###### Motivation for this change

[Marwaita](https://www.pling.com/p/1239855/) is a GTK theme supporting Budgie, Pantheon, Mate and Xfce4 desktops.

4dd0f5eb29247bc0d15d56e66cc8dce640b7 (1)

08547d71610893d43db3795c477e79d31829 (1)

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by timokau** [comment]
Bot mistake.
/status needs_reviewer

File: pkgs/data/themes/marwaita/default.nix
Comment (fgaz) [inline pkgs/data/themes/marwaita/default.nix:11]: `"unstable-2020-07-01"`, since this isn't an actual version

**Comment by fgaz** [comment]
Also I don't use any of those desktops and I don't know anyone who does, so I'm bouncing this to someone else for testing

/status needs_reviewer

**Comment by symphorien** [comment]
tested on xfce.
/status needs_merge

File: pkgs/data/themes/marwaita/default.nix
Comment (romildo) [reply pkgs/data/themes/marwaita/default.nix:11]: As far as I know adding _unstable_ to the name is not mandatory. There are some discussions on how it should be done (but I am failing to find the link now). It would be part of _pname_ or _version_? It should appear before or after the date? I do not like it written just before the date because it could be understood as part of _pname_, while it really is not (or is it?). I prefer not using _unstable_ on packages without an explicit need to distinguish from stable versions simultaneously available in nixpkgs.

Comment (timokau) [reply pkgs/data/themes/marwaita/default.nix:11]: I don't really have a horse in this race, but the [guideline](https://nixos.org/nixpkgs/manual/#sec-package-naming) is pretty clear:

> If a package is not a release but a commit from a repository, then the version part of the name must be the date of that (fetched) commit. The date must be in "YYYY-MM-DD" format. Also append "unstable" to the name - e.g., "pkgname-unstable-2014-09-23".

That is, `pname` should have an "-unstable" suffix. I can see at least one advantage: It makes it possible to distinguish between packages that actually name their releases after dates and packages without releases where we just decided to ship some commit.

**Comment by timokau** [comment]
> Also I don't use any of those desktops and I don't know anyone who does, so I'm bouncing this to someone else for testing

This might be a bit controversial, but I would say runtime testing is at most a "nice to have" for a review. In my mind, the reviewers job is to check that the actual diff is good and does not obviously break anything (i.e. breaks at build or test time). For the runtime testing its sufficient to trust the PR author.

Also if you *do* want find someone who could test this, it would be preferable to actively look someone up (for example with `git log`) and ping them. Setting it back to `needs_reviewer` will only assign some other random reviewer which likely doesn't have the needed expertise / setup either.

Anyway, thanks for the reviews @fgaz and @symphorien :)

File: pkgs/data/themes/marwaita/default.nix
Comment (romildo) [reply pkgs/data/themes/marwaita/default.nix:11]: There are plenty of discussions regarding this use of _unstable_. In particular [this comment](https://github.com/NixOS/nixpkgs/pull/88023#issuecomment-629870275) express my thinking about it.

Some links:
https://github.com/NixOS/nixpkgs/pull/88023
https://github.com/NixOS/nixpkgs/issues/68518
https://github.com/NixOS/nixpkgs/pull/86430

Comment (timokau) [reply pkgs/data/themes/marwaita/default.nix:11]: Thanks for linking those! As I said, I do think it makes *some* sense, but I don't really have a horse in the race. The PR looks fine otherwise, since you have merge permission yourself I'll let you make the final call.

**Review by timokau — APPROVED** [review]

**Comment by fgaz** [comment]
> For the runtime testing its sufficient to trust the PR author.

In most cases including this one maybe yes, but what about new authors and init prs?

**Comment by timokau** [comment]
> > For the runtime testing its sufficient to trust the PR author.
> 
> In most cases including this one maybe yes, but what about new authors and init prs?

Init PRs are very low-stake: The worst that's going to happen is that we have a new not-quite-working package in the repository. Basically the author has no incentive to do that on purpose, so its relatively unlikely. They will have tested it at least for their use-case. If it does happen, someone is going to notice at some point and fix it / disable it. Either way the worst-case is both "not so bad" and "pretty unlikely".

New authors (or anyone) making changes to existing things might be a bit more tricky, since here the worst case is that it breaks someones setup and the testing burden is "any usecase some nixpkgs user might currently have for the package" which the PR author probably didn't test for. Still, I'd argue that static checks (i.e. builds fine and tests pass) are sufficient in most cases. If something does break on runtime, that's why the channel is called unstable. Its a motivation for the package maintainer to add more/better static checks.

This basically goes back to my personal definition of the unstable channel: We have tested that nothing obvious breaks (`nix-review`), but things might break on runtime. That's similar to what most other distros call "unstable", with the added benefit of easy rollbacks.

**Pull request merged**
